### DISCUSSION TITLE
[SOLVED] Crash in `SignpostReducer` when using `_printChanges...` in `prerelease/1.0` (I'm using the new navigation apis.

### DISCUSSION THREAD

**Discussion by oliverfoggin** [pointfreeco/swift-composable-architecture#1953] [Beta]
Screenshot of where the crash is. Line 147 of `SignpostReducer`

<img width="1292" alt="image" src="https://user-images.githubusercontent.com/1813244/222092043-7c25033a-7de6-4044-a452-8c098b83f86a.png">

Crash log...

>Thread 13: EXC_BAD_ACCESS (code=1, address=0x8bc0ce852c80)

When I remove the `_printChanges` completely this doesn't happen.

I have recently updated one of the `Reducer`s in our app to change it to using a `@PresentationState var destination` enum. To improve the modelling of the app.

I have three destination cases...

```
public enum Destination: Equatable {
	case addSave(EditSave.State)
	case editSave(EditSave.State)
	case alert(AlertState<Action.Alert>)
}
```

Which are scoped to like...

```
// Parent Reducer
.ifLet(\.$destination, action: /Action.destination) {
	Scope(state: /Destination.addSave, action: /Action.Destination.addSave) {
		EditSave()
	}
	Scope(state: /Destination.editSave, action: /Action.Destination.editSave) {
		EditSave()
	}
	Scope(state: /Destination.alert, action: /Action.Destination.alert) {
		EmptyReducer()
	}
}
```

In the view these are navigated to like...

```
.sheet(
	store: store.scope(state: \.$destination, action: RecurringSavesFeature.Action.destination),
	state: /RecurringSavesFeature.Destination.addSave,
	action: RecurringSavesFeature.Action.Destination.addSave
) {
	EditSaveView(store: $0)
}
.sheet(
	store: store.scope(state: \.$destination, action: RecurringSavesFeature.Action.destination),
	state: /RecurringSavesFeature.Destination.editSave,
	action: RecurringSavesFeature.Action.Destination.editSave
) {
	EditSaveView(store: $0)
}
.alert(
	store: store.scope(state: \.$destination, action: RecurringSavesFeature.Action.destination),
	state: /RecurringSavesFeature.Destination.alert,
	action: RecurringSavesFeature.Action.Destination.alert
)
```

When I navigate to either of the sheets the sheet loads and then this error happens.

**Comment by tgrapperon** [comment]
Hey @oliverfoggin! Before diving into it, have you tried to rebuild the app from a clean slate? There are currently issues with incremental builds in Xcode (#1742).

**Reply by oliverfoggin** [reply]
I will try now. Thanks

**Reply by oliverfoggin** [reply]
Well... that seems to have resolved it. Thanks 😅
